src/meson.build

```diff
@@ -680,11 +673,7 @@ if get_option('tests').enabled()
   env.set('HBHEADERS', ' '.join(hb_headers))
 
   if cpp.get_id() != 'msvc' and not meson.is_cross_build() # ensure the local tools are usable
-    # See https://github.com/mesonbuild/meson/pull/6838
-    if meson.version().version_compare('>=0.54.999') and not get_option('with_libstdcxx')
-      dist_check_script += 'check-libstdc++'
-    endif
-    dist_check_script += ['check-static-inits', 'check-symbols']
+    dist_check_script += ['check-libstdc++', 'check-static-inits', 'check-symbols']
```

Comment:
Why I am not even bothering checking `>=0.54.999` for link_language feature to be added to meson? Because meson `-Wl,--as-needed` which make this pass even if a cxx linker is used.